src/magicbox/ResultPreviewsManager.ts

```diff
@@ -203,8 +203,10 @@ export class ResultPreviewsManager {
     return this.previewsProcessor.processQueries(populateEventArgs.previewsQueries);
   }
 
-  private updateSearchResultPreviewsHeader(text: string) {
-    this.resultPreviewsHeader.text(text);
+  private updateSearchResultPreviewsHeader(suggestion: string) {
+    this.resultPreviewsHeader.setHtml(
```

Comment:
It's risky to use `setHtml` on dynamic values because someone might abuse it to do a XSS attack. Now if we can bold the entire text, we shouldn't need to use a span tag and `setHtml`, so we can avoid the vulnerability.

If the design was changed though, then we will need to sanitize the string for dangerous characters. We will also need to add a unit test passing in a html tag and assert that it gets sanitized.

Reply:
yes good point. At least please add a unit test with an XSS attack inside it to check if it's cleared correctly. There is a few unit tests like this in the code base, Ctrl+f for "XSS"

Reply:
This was only so a specific piece of text could be bolded, given that now it's all bold, a `setHtml` won't be necessary anymore. But I'll remember to add unit tests if risks of XSS injections come up again.